smfile_parser.py

```diff
@@ -1,103 +1,95 @@
 
     with open(sm_file, encoding='ascii', errors='ignore') as f:
         for line in f:
             if not flag:
                 if line.startswith('#TITLE:'):
-                    step_dict['title']  = line.lstrip('#TITLE').lstrip(':').rstrip(';\n')
+                    data['title'] = line
                 elif line.startswith('#BPMS:'):
-                    if ',' in line:  # raises Exception if multiple BPMS detected
+                    if ',' in line:
                         raise ValueError('Multiple BPMs detected')
-                    step_dict['BPM']    = float(line.lstrip('#BPMS:0.0').lstrip('0').lstrip('=').rstrip(';\n'))
+                    data['bpm'] = line
                 elif line.startswith('#OFFSET:'):
-                    step_dict['offset'] = float(line.lstrip('#OFFSET').lstrip(':').rstrip(';\n'))
-                elif line.startswith('#STOPS:') and line.rstrip('\n') != "#STOPS:;":
+                    data['offset'] = line
+                elif line.startswith('#STOPS') and line.rstrip('\n') != '#STOPS:;':
                     raise ValueError('Stop detected')
-
-                elif line.startswith('#NOTES:'):
-                    flag     = True
-                    next(f)
-                    next(f)
-                    step_dict['types'].append("DIFFICULTY ")
-                    step_dict['notes'].append(next(f).lstrip(' ').rstrip(':\n'))
-
-            else:   #start of note processing
-                if line[0].isdigit():
-                    check = True if any((c in set('123456789')) for c in line) else False
-                    if check:
-                        measure.append(True)
-                        step_dict['types'].append(convert_note(line.rstrip('\n')))
-                    else:
-                        measure.append(False)
-                elif line.startswith(',') or line.startswith(';'):
-                    line_timing = calculate_timing(measure, measure_index, step_dict['BPM'], step_dict['offset'])
-                    step_dict['notes'].extend(line_timing)
-                    measure.clear()
-                    if line.startswith(';'):
-                        output_file(new_file, step_dict, output_dir)
-                    measure_index += 1
                 elif line.startswith('#NOTES:'):
-                    measure_index = 0                   
+                    flag = True
+            if flag:
+                if not (line.startswith(' ') or line.startswith('/') or line.startswith('\n') or line.startswith('#')):
+                    data['notes'].append(sub('4', '1', sub('[MKLF]', '0', line)).rstrip('\n'))
+                if line.startswith('#NOTES:'):
                     next(f)
                     next(f)
-                    step_dict['types'].append("DIFFICULTY ")
-                    step_dict['notes'].append(next(f).lstrip(' ').rstrip(':\n'))
-    return step_dict
+                    data['notes'].append('DIFFICULTY ' + next(f).lstrip(' ').rstrip(':\n'))
```

Comment:
This breaks the format of parsing 1 line per loop. Can this be changed to keep that logic?